### DISCUSSION TITLE
Subsample fragments in the ArchRobject?

### DISCUSSION THREAD

**Discussion by yding9** [GreenleafLab/ArchR#808] [Questions / Documentation]
Hello Jeffrey and Ryan,

The following is my task: I have 5k cells in one ArchRobject with different fragments per cell. Is there a simple way to subsample fragments per cell to the same/similar number (e.g. 10,000 fragments per cell)? I need to control fragments number per cell for a well-controlled test. 

My native thought is to export fragment/bed files from each library, sort files based on cell barcodes, subsample fragments, merge the fragments files for each library, and recreate arrow files from subsampled fragments files. But this doesn't look like a smart solution. I'm wondering if there is some workaround that allows us to subsample fragments in the ArchRobject directly?

Thank you so much for this wonderful tool and your help in advance!

**Comment by rcorces** [comment]
To the best of my knowledge, there is no easy way to do this within ArchR. I think your proposed solution is probably best.

**Reply by yding9** [reply]
OK, good to know! Thank you so much for your quick response, Ryan!
